### PR TITLE
[Broker] no-persistent topic support terminate

### PR DESCRIPTION
**Pull request by shibd**
#11834

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java [hunk 1]
```diff
@@ -328,6 +329,16 @@ public void removeProducer(Producer producer) {
         return delete(false, false, false);
     }
 
+    @Override
+    public CompletableFuture<MessageId> terminate() {
+        CompletableFuture<MessageId> future = new CompletableFuture<>();
+        producers.values().forEach(Producer::disconnect);
```
Comment (congbobo184) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java:335]: why not disconnect sub?

**Review by congbobo184 — CHANGES_REQUESTED** [review]
Could you please add a test for it?

**Comment by shibd** [comment]
> Could you please add a test for it?

Ok, I'll add it

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java (hunk 1 above)
Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java:335]: do we need retain existing consumer when non-persistent topic terminate? @gaoran10 @codelipenghui /cc

Comment (gaozhangmin) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentTopic.java:335]: I think non-persistent termination should behave like persistent

**Review by merlimat — CHANGES_REQUESTED** [review]
The problem with non-persistent topics is that the termination will be lost at the next broker restart or topic failover so it would not make much practical sense.

**Comment by shibd** [comment]
> The problem with non-persistent topics is that the termination will be lost at the next broker restart or topic failover so it would not make much practical sense.

Got it, I will close this pr.

**Pull request closed without merging**